cassandra-analytics-integration-framework/src/main/java/org/apache/cassandra/sidecar/testing/IntegrationTestBase.java

```diff
@@ -333,4 +267,70 @@ protected void waitForKeyspaceAndTable(String keyspaceName, String tableName)
         String.format("Keyspace/table %s/%s did not become visible on all sidecar instances",
                       keyspaceName, tableName));
     }
+
+    /**
+     * A {@link DnsResolver} instance used for tests that provides fast DNS resolution, to avoid blocking
+     * DNS resolution at the JDK/OS-level.
+     *
+     * <p><b>NOTE:</b> The resolver assumes that the addresses are of the form 127.0.0.x, which is what is currently
+     * configured for integration tests.
+     */
+    static class FastDnsResolver implements DnsResolver
+    {
+        private static final Logger LOGGER = LoggerFactory.getLogger(FastDnsResolver.class);
+        private static final Pattern HOSTNAME_PATTERN = Pattern.compile("^localhost(\\d+)?$");
+
+        /**
+         * Returns the resolved IP address from the hostname.
+         *
+         * <pre>
+         * resolver.resolve("localhost") = "127.0.0.1"
+         * resolver.resolve("localhost2") = "127.0.0.2"
+         * resolver.resolve("localhost20") = "127.0.0.20"
+         * resolver.resolve("127.0.0.5") = "127.0.0.5"
+         * </pre>
+         *
+         * @param hostname the hostname to resolve
+         * @return the resolved IP address
+         */
+        @Override
+        public String resolve(String hostname)
+        {
+            Matcher matcher = HOSTNAME_PATTERN.matcher(hostname);
+            if (!matcher.matches())
+            {
+                LOGGER.warn("Invalid hostname found {}. Skipping resolution", hostname);
+                return hostname;
```

Comment:
Instead of assuming defaults, can this fallback to the default resolver? That way, it will take advantage of our optimization and also work in any custom dev environments.

Same comment for line306 and in `reverseResolve`.  (We will likely need a WARN log for that case as well)

Reply:
good idea, I'll push the change in a bit